querqy-core/src/main/java/querqy/rewrite/contrib/ReplaceRewriter.java

```diff
@@ -0,0 +1,111 @@
+package querqy.rewrite.contrib;
+
+import querqy.ComparableCharSequence;
+import querqy.CompoundCharSequence;
+import querqy.LowerCaseCharSequence;
+import querqy.model.AbstractNodeVisitor;
+import querqy.model.Clause;
+import querqy.model.DisjunctionMaxQuery;
+import querqy.model.ExpandedQuery;
+import querqy.model.Node;
+import querqy.model.Query;
+import querqy.model.Term;
+import querqy.rewrite.QueryRewriter;
+import querqy.trie.State;
+import querqy.trie.TrieMap;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static querqy.rewrite.contrib.ReplaceRewriterParser.TOKEN_SEPARATOR;
+
+public class ReplaceRewriter extends AbstractNodeVisitor<Node> implements QueryRewriter {
+
+    private final TrieMap<List<ComparableCharSequence>> replaceRules;
```

Comment:
Can we just use the more general CharSequence instead of ComparableCharSequence here?